profiling/src/php_ffi.c

```diff
@@ -0,0 +1,210 @@
+#include "php_ffi.h"
+
+#include <assert.h>
+#include <stdbool.h>
+#include <string.h>
+
+const char *datadog_extension_build_id(void) { return ZEND_EXTENSION_BUILD_ID; }
+const char *datadog_module_build_id(void) { return ZEND_MODULE_BUILD_ID; }
+
+ZEND_DECLARE_MODULE_GLOBALS(datadog_php_profiling)
+
+#ifdef ZTS
+#define DATADOG_PHP_PROFILING_G(v) ZEND_MODULE_GLOBALS_ACCESSOR(datadog_php_profiling, v)
+#else
+#define DATADOG_PHP_PROFILING(v) (datadog_php_profiling_globals.v)
+#endif
+
+#if PHP_VERSION_ID >= 80000
+#define DD_INI_GET_ADDR() ZEND_INI_GET_ADDR()
+#else
+#ifndef ZTS
+#define DD_INI_GET_BASE() ((char *)mh_arg2)
+#else
+#define DD_INI_GET_BASE() ((char *)ts_resource(*((int *)mh_arg2)))
+#endif
+#define DD_INI_GET_ADDR() (DD_INI_GET_BASE() + (size_t)mh_arg1)
+#endif
+
+#if PHP_VERSION_ID < 70200
+static zend_string *zend_string_init_interned(const char *str, size_t len, int persistent) {
+    zend_string *ret = zend_string_init(str, len, persistent);
+
+    return zend_new_interned_string(ret);
+}
+#endif
+
+/**
+ * Returns true if `a` and `b` have the same size and all their bytes commpare
+ * as equal. The pointers `a.ptr` and `b.ptr` should not be NULL (please use an
+ * empty string).
+ */
+static bool str_eq(datadog_php_str a, datadog_php_str b) {
+    assert(a.ptr);
+    assert(b.ptr);
+    return a.size == b.size && memcmp(a.ptr, b.ptr, b.size) == 0;
+}
+
+/**
+ * Converts the ASCII char to its lowercase equivalent if it's a recognized
+ * uppercase character; otherwise returns the char without modification.
+ */
+static char char_tolower_ascii(char c) { return (char)(c >= 'A' && c <= 'Z' ? (c - ('A' - 'a')) : c); }
+
+static void str_copy_tolower_ascii(size_t len, const char src[static len], char dest[static len]) {
+    for (size_t i = 0; i != len; ++i) {
+        *(dest++) = char_tolower_ascii(src[i]);
+    }
+}
+
+static zend_result datadog_php_profiling_log_level_parse(datadog_php_str str, uintptr_t *log_level) {
```

Comment:
I'm not seeing anywhere this is used other than the test for it

Reply:
Yes, I think so. It used to be used in the `.ini` parsing bits.